src/app/3d/qgs3dmaptoolpointcloudchangeattributepolygon.cpp

```diff
@@ -27,78 +27,106 @@
 #include <QApplication>
 #include <QMouseEvent>
 
-
-Qgs3DMapToolPointCloudChangeAttributePolygon::Qgs3DMapToolPointCloudChangeAttributePolygon( Qgs3DMapCanvas *canvas )
-  : Qgs3DMapToolPointCloudChangeAttribute( canvas )
+Qgs3DMapToolPointCloudChangeAttributePolygon::Qgs3DMapToolPointCloudChangeAttributePolygon( Qgs3DMapCanvas *canvas, const ToolType type = Polygon )
+  : Qgs3DMapToolPointCloudChangeAttribute( canvas ), mToolType( type )
 {
 }
 
 Qgs3DMapToolPointCloudChangeAttributePolygon::~Qgs3DMapToolPointCloudChangeAttributePolygon() = default;
```

Comment:
shouldn't we delete the line and polygon rubber bands in the destructor? otherwise they will probably stay in the scene?